### PR TITLE
Added linux-gcc-5-qt-{4,5} & linux-clang-3.9-node-4

### PR DESCRIPTION
**Pull request by brunoabinader**
This PR:
- Updates revision from 2 to 3:
  - `base` now inherits from `libsysconfcpus` - needed for being able to find the shared object file
  - Added `locales` debian package as dependency for `base` - needed when issuing `locale-gen`
  - Added `lcov` debian package as dependency for `linux`
- Adds the following images:
  - linux-gcc-5-qt-4
  - linux gcc-5-qt-5
  - linux-clang-3.9-node-4

### CODE REVIEW

**Comment by kkaefer** [comment]
> `base` now inherits from `libsysconfcpus` - needed for being able to find the shared object file

`libsysconfcpus` only exists as a compilation image; I retrieved the compiled .so file and it should be added to the `base` image. Existing images were able to find `libsysconfcpus.so`. What prompted you to change this?

**Comment by kkaefer** [comment]
Sorry, I just realized that I have an ignore set on .so files, so it was never committed. Will add now.

File: all.sh [hunk 1]
```diff
@@ -12,6 +12,10 @@ docker build -t mbgl/ci:latest-java --squash java
 # Final images
 docker build -t mbgl/ci:r${REVISION}-linux-clang-3.9 --squash linux-clang-3.9
 docker push mbgl/ci:r${REVISION}-linux-clang-3.9
+docker tag mbgl/ci:r${REVISION}-linux-clang-3.9 mbgl/ci:latest-linux-clang-3.9
```
Comment (kkaefer) [inline all.sh]: Can you fold this into the `build` step above? We can add two `-t` flags.

File: linux-gcc-5-qt-5/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+RUN set -eu \
+ && make \
+ && make install
+
+ENV PATH="/opt/qt-5.9/bin:$PATH"
+
+WORKDIR /opt
+
+RUN set -eu \
+ && rm -rf qt5
```
Comment (kkaefer) [inline linux-gcc-5-qt-5/Dockerfile]: Can we also run `rm -rf /var/lib/apt/lists/*`?

File: linux-gcc-5-qt-5/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    --module-subset=qtbase,qtxmlpatterns,qtdeclarative,qtgraphicaleffects,qtquickcontrols,qtsvg,qtimageformats,qtlocation,qtserialport,qttools
+
+WORKDIR /opt/qt5/build
+
+RUN set -eu \
+ && mkdir -p /opt/qt-5.9 \
+ && ../configure \
+    -opensource -confirm-license \
+    -nomake tests -nomake examples \
+    -debug -prefix /opt/qt-5.9 \
```
Comment (kkaefer) [inline linux-gcc-5-qt-5/Dockerfile]: Does that build the debug version of Qt or include debug symbols?

Comment (brunoabinader) [inline linux-gcc-5-qt-5/Dockerfile]: It does both - we could however replace that with a release build and use `-force-debug-info`, which create symbol files for release builds, though I have never tested that.

Comment (brunoabinader) [inline linux-gcc-5-qt-5/Dockerfile]: In the end I'd guess the result is the same - the size of the separate symbol files would be the same as having them included on the binaries.

Comment (kkaefer) [inline linux-gcc-5-qt-5/Dockerfile]: Okay, but given this is for testing, building a debug version of Qt is fine, I assume?

File: linux-gcc-5-qt-5/Dockerfile (hunk 2 above)
Comment (brunoabinader) [inline linux-gcc-5-qt-5/Dockerfile]: Done.

File: linux-gcc-5-qt-5/Dockerfile (hunk 3 above)
Comment (brunoabinader) [inline linux-gcc-5-qt-5/Dockerfile]: Yeah - we need it e.g. when generating the valgrind backtraces.

**Comment by brunoabinader** [comment]
Succeeded building all images locally 👍

File: linux-gcc-5-qt-5/Dockerfile (hunk 3 above)
Comment (brunoabinader) [inline linux-gcc-5-qt-5/Dockerfile]: It turns out having a debug Qt5 build had big performance implications when running memory load tests under valgrind - I was obtaining CI runs up to 23 minutes long. Switching the Qt5 build to release decreased the docker image size from 3G to 1.5G (uncompressed) and also made it possible to switch the amount of jobs back to 6 (used by other jobs as default).

**Pull request merged**